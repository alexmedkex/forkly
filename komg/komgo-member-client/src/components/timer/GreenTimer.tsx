import * as React from 'react'
import { Moment } from 'moment'
import { Icon } from 'semantic-ui-react'
import { printExpiryDateInForm } from '../../utils/timer'

interface IProps {
  dueDateMoment: Moment
  leftMinutes: number
}

const GreenTimer: React.FC<IProps> = (props: IProps) => {
  const { dueDateMoment } = props
  return (
    <React.Fragment>
      <Icon name="clock outline" color="green" />
      <span className="grey" data-test-id="timer-more-than-24h">
        {printExpiryDateInForm(dueDateMoment)}
      </span>
    </React.Fragment>
  )
}

export default GreenTimer
