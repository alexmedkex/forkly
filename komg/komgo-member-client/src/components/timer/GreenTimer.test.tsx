import * as React from 'react'
import * as renderer from 'react-test-renderer'
import moment from 'moment-timezone'
import GreenTimer from './GreenTimer'

describe('GreenTimer component', () => {
  let defaultProps
  beforeEach(() => {
    moment.tz.guess = jest.fn(() => 'Europe/Belgrade')
    moment.tz.setDefault('Europe/Belgrade')
    Date.now = jest.fn(() => 1487076708000)
    defaultProps = {
      leftMinutes: 2000,
      dueDateMoment: moment(1553431046787 + 120000)
    }
  })

  afterEach(() => {
    moment.tz.setDefault()
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<GreenTimer {...defaultProps} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
