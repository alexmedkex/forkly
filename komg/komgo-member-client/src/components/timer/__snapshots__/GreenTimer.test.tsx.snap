// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreenTimer component should match snapshot 1`] = `
Array [
  <i
    aria-hidden="true"
    className="green clock outline icon"
  />,
  <span
    className="grey"
    data-test-id="timer-more-than-24h"
  >
    2019/03/24, Sunday, 1:39 PM CET
  </span>,
]
`;
